cirq/ops/raw_types_test.py

```diff
@@ -459,6 +459,49 @@ def test_tagged_operation():
     assert op.with_qubits(q2).qubits == (q2,)
 
 
+def test_circuit_diagram():
+    h = cirq.H(cirq.GridQubit(1, 1))
+    tagged_h = h.with_tags('tag1')
+    expected = cirq.CircuitDiagramInfo(wire_symbols=("H['tag1']",),
+                                       exponent=1.0,
+                                       connected=True,
+                                       exponent_qubit_index=None,
+                                       auto_exponent_parens=True)
+    args = cirq.CircuitDiagramInfoArgs(None, None, None, None, None, False)
+    assert cirq.circuit_diagram_info(tagged_h) == expected
+    assert (cirq.circuit_diagram_info(tagged_h,
+                                      args) == cirq.circuit_diagram_info(h))
+
+    c = cirq.Circuit(tagged_h)
+    diagram_with_tags = "(1, 1): ───H['tag1']───"
+    diagram_without_tags = "(1, 1): ───H───"
+    assert str(cirq.Circuit(tagged_h)) == diagram_with_tags
+    assert c.to_text_diagram() == diagram_with_tags
+    assert c.to_text_diagram(include_tags=False) == diagram_without_tags
+
+
+def test_circuit_diagram_corner_cases():
+    global_phase = cirq.GlobalPhaseOperation(coefficient=1.0).with_tags('tag0')
+    assert global_phase._circuit_diagram_info_(args=None) == NotImplemented
```

Comment:
This should test `cirq.circuit_diagram_info(globa_phase) == ?`

Reply:
This is tricky, since that just raises a TypeError.  Even though this accesses a private method, I felt like it was a better test, since it tests the forwarding.  We could test that the cirq.circuit_diagram_info(global_phase) raises TypeError, but that doesn't feel right to me, since there's a bunch of ways this can throw Type Error.

Reply:
Then test that `cirq.circuit_diagram_info(..., default='default') == 'default'`.